sycl/doc/design/CommandGraph.md

```diff
@@ -1,66 +1,164 @@
-| `piextCommandBufferRetain` | incrementing reference count of command-buffer. |
-| `piextCommandBufferRelease` | decrementing reference count of command-buffer. |
-| `piextCommandBufferFinalize` | no more commands can be appended, makes command buffer ready to enqueue on command-queue. |
-| `piextCommandBufferNDRangeKernel` | append a kernel execution command to command buffer. |
-| `piextEnqueueCommandBuffer` | submit command-buffer to queue for execution |
-| `piextCommandBufferMemcpyUSM` | append a USM memcpy command to the command-buffer. |
-| `piextCommandBufferMemBufferCopy` | append a mem buffer copy command to the command-buffer. |
-| `piextCommandBufferMemBufferCopyRect` | append a rectangular mem buffer copy command to the command-buffer. |
+command-list multiple times.
+
+### UR Command-Buffer Experimental Feature
+
+The command-buffer concept has been introduced to UR as an
+[experimental feature](https://oneapi-src.github.io/unified-runtime/core/api.html#command-buffer-experimental)
+with the following entry-points:
+
+| Function                                | Description |
+| --------------------------------------- | ----------- |
+| `urCommandBufferCreateExp`              | Create a command-buffer. |
+| `urCommandBufferRetainExp`              | Incrementing reference count of command-buffer. |
+| `urCommandBufferReleaseExp`             | Decrementing reference count of command-buffer. |
+| `urCommandBufferFinalizeExp`            | No more commands can be appended, makes command-buffer ready to enqueue on a command-queue. |
+| `urCommandBufferAppendKernelLaunchExp`  | Append a kernel execution command to command-buffer. |
+| `urCommandBufferAppendMemcpyUSMExp`     | Append a USM memcpy command to the command-buffer. |
+| `urCommandBufferAppendMembufferCopyExp` | Append a mem buffer copy command to the command-buffer. |
+| `urCommandBufferEnqueueExp`             | Submit command-buffer to a command-queue for execution. |
+
+See the [UR EXP-COMMAND-BUFFER](https://oneapi-src.github.io/unified-runtime/core/EXP-COMMAND-BUFFER.html)
+specification for more details.
 
 ## Design
 
 ![Basic architecture diagram.](images/SYCL-Graph-Architecture.svg)
 
 There are two sets of user facing interfaces that can be used to create a
-command graph object: 
-Explicit and Record & Replay API. Within the runtime they share a common
-infrastructure.
+command-graph object: Explicit and Record & Replay API. Within the runtime they
+share a common infrastructure.
+
+## Nodes & Edges
+
+Nodes are represented by an internal `detail::node_impl` class which
+stores a `sycl::detail::CG` object for the command-group that the node
+represents. When a user adds a node to a graph using the explicit
+`command_graph<modifiable>::add()` API passing a CGF, in the implementation a
+`sycl::handler` object is created which finalizes the CGF, and after
+finalization the CG object from the handler is moved to the node.
```

Comment:
I think it could be worth clarifying in this section that we're preventing the `CG` object from being submitted for execution as normal inside `handler::finalize()` and then storing it in the graph as a new node instead.

Reply:
Done, i've also expanded this section quite a bit